src/nvim/buffer_defs.h

```diff
@@ -587,27 +587,27 @@ struct file_buffer {
   int b_p_pi;                   ///< 'preserveindent'
   char *b_p_qe;                 ///< 'quoteescape'
   int b_p_ro;                   ///< 'readonly'
-  long b_p_sw;                  ///< 'shiftwidth'
-  long b_p_scbk;                ///< 'scrollback'
+  OptInt b_p_sw;                ///< 'shiftwidth'
+  OptInt b_p_scbk;              ///< 'scrollback'
   int b_p_si;                   ///< 'smartindent'
-  long b_p_sts;                 ///< 'softtabstop'
-  long b_p_sts_nopaste;         ///< b_p_sts saved for paste mode
+  OptInt b_p_sts;               ///< 'softtabstop'
+  OptInt b_p_sts_nopaste;       ///< b_p_sts saved for paste mode
   char *b_p_sua;                ///< 'suffixesadd'
   int b_p_swf;                  ///< 'swapfile'
-  long b_p_smc;                 ///< 'synmaxcol'
+  OptInt b_p_smc;               ///< 'synmaxcol'
   char *b_p_syn;                ///< 'syntax'
-  long b_p_ts;                  ///< 'tabstop'
-  long b_p_tw;                  ///< 'textwidth'
-  long b_p_tw_nobin;            ///< b_p_tw saved for binary mode
-  long b_p_tw_nopaste;          ///< b_p_tw saved for paste mode
-  long b_p_wm;                  ///< 'wrapmargin'
-  long b_p_wm_nobin;            ///< b_p_wm saved for binary mode
-  long b_p_wm_nopaste;          ///< b_p_wm saved for paste mode
+  OptInt b_p_ts;                ///< 'tabstop'
+  OptInt b_p_tw;                ///< 'textwidth'
+  OptInt b_p_tw_nobin;          ///< b_p_tw saved for binary mode
+  OptInt b_p_tw_nopaste;        ///< b_p_tw saved for paste mode
+  OptInt b_p_wm;                ///< 'wrapmargin'
+  OptInt b_p_wm_nobin;          ///< b_p_wm saved for binary mode
+  OptInt b_p_wm_nopaste;        ///< b_p_wm saved for paste mode
   char *b_p_vsts;               ///< 'varsofttabstop'
-  long *b_p_vsts_array;         ///< 'varsofttabstop' in internal format
+  OptInt *b_p_vsts_array;       ///< 'varsofttabstop' in internal format
```

Comment:
this is not directly an option but state we parsed from an option, therefore it could have any type of our choosing (it was a mistake to make this a `long` as if it was an option in the first place, but no need to repeat the mistake), so it could just be `int *` or even `colnr_T *`